### PR TITLE
[SPARK-35040][PYTHON] Remove Spark-version related codes from test codes

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?

Removes PySpark version dependent codes from pyspark.pandas test codes.

### Why are the changes needed?

There are several places to check the PySpark version and switch the logic, but now those are not necessary.
We should remove them.

We will do the same thing after we finish porting tests.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Existing tests.

Keyword:SPARK-34849

### CODE REVIEW

File: python/pyspark/pandas/tests/indexes/test_base.py [hunk 1]
```diff
@@ -2249,13 +2235,7 @@ def test_to_list(self):
...
 
         self.assert_eq(kidx.tolist(), pidx.tolist())
-
-        if LooseVersion(pyspark.__version__) < LooseVersion("2.4"):
-            # PySpark < 2.4 does not support struct type with arrow enabled.
-            with self.sql_conf({SPARK_CONF_ARROW_ENABLED: False}):
-                self.assert_eq(kmidx.tolist(), pmidx.tolist())
-        else:
-            self.assert_eq(kidx.tolist(), pidx.tolist())
+        self.assert_eq(kidx.tolist(), pidx.tolist())
```
Comment (ueshin) [inline python/pyspark/pandas/tests/indexes/test_base.py:2238]: Seems like the same tests?
I guess the second one should be `kmidx.tolist()`?

File: python/pyspark/pandas/tests/test_series.py [hunk 2]
```diff
@@ -857,15 +848,7 @@ def _test_value_counts(self):
...
-        if LooseVersion(pyspark.__version__) < LooseVersion("2.4"):
-            with self.sql_conf({SPARK_CONF_ARROW_ENABLED: False}):
-                self._test_value_counts()
-            self.assertRaises(
-                RuntimeError,
-                lambda: ps.MultiIndex.from_tuples([("x", "a"), ("x", "b")]).value_counts(),
-            )
-        else:
-            self._test_value_counts()
+        self._test_value_counts()
```
Comment (ueshin) [inline python/pyspark/pandas/tests/test_series.py:851]: Do we call `self._test_value_counts()` somewhere else?
If not, we can just remove this and rename `_test_value_counts` to `test_value_counts`?

File: python/pyspark/pandas/tests/indexes/test_base.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/indexes/test_base.py]: Good catch! Updated.

File: python/pyspark/pandas/tests/test_series.py (hunk 2 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_series.py]: `_test_value_counts` is called just once. It is renamed to `test_value_counts`.

**Comment by ueshin** [comment]
Jenkins, retest this please.

**Review by ueshin — APPROVED** [review]
LGTM, pending tests.

**Comment by ueshin** [comment]
Thanks! merging to master.

**Pull request closed without merging**